jetstream/experimenter.py

```diff
@@ -165,24 +168,55 @@ class ExperimentCollection:
     @classmethod
     def from_experimenter(cls, session: requests.Session = None) -> "ExperimentCollection":
         session = session or requests.Session()
-        legacy_experiments_json = session.get(cls.EXPERIMENTER_API_URL_V1).json()
-        legacy_experiments = []
 
-        for experiment in legacy_experiments_json:
-            if experiment["type"] != "rapid":
-                try:
-                    legacy_experiments.append(ExperimentV1.from_dict(experiment).to_experiment())
-                except Exception as e:
-                    logger.exception(str(e), exc_info=e, extra={"experiment": experiment["slug"]})
+        v1_retries = 0
+        success_v1 = False
+        v6_retries = 0
+        success_v6 = False
 
-        nimbus_experiments_json = session.get(cls.EXPERIMENTER_API_URL_V6).json()
-        nimbus_experiments = []
-
-        for experiment in nimbus_experiments_json:
+        while v1_retries < cls.MAX_RETRIES and not success_v1:
+            try:
+                legacy_experiments_json = session.get(cls.EXPERIMENTER_API_URL_V1).json()
+                legacy_experiments = []
+                success_v1 = True
+
+                for experiment in legacy_experiments_json:
+                    if experiment["type"] != "rapid":
+                        try:
+                            legacy_experiments.append(
+                                ExperimentV1.from_dict(experiment).to_experiment()
+                            )
+                        except Exception as e:
+                            logger.exception(
+                                str(e), exc_info=e, extra={"experiment": experiment["slug"]}
+                            )
```

Comment:
we can move all of this out of the while/try/except and just leave the get(...).json() call inside, which will at least let us dedent a little.

you could use this as an excuse to bust out a for/else if you want:

```python
def retry_get(session, url, retries) -> Mapping[str, Any]:
  for _i in range(retries):
    try:
      blob = session.get(url).json()
      break
    except Exception:
      logger.info("what the heck")
      time.sleep(1)
  else:
    raise Exception("too many retries")
  return blob
```

Reply:
That's way better :)